shell/platform/android/io/flutter/embedding/engine/android/AndroidTouchProcessor.java

```diff
@@ -0,0 +1,186 @@
+      // but it's the responsibility of a later part of the system to
+      // ignore 0-deltas if desired.
+      for (int p = 0; p < pointerCount; p++) {
+        addPointerForIndex(event, p, packet);
+      }
+    }
+
+    // Verify that the packet is the expected size.
+    assert packet.position() % (POINTER_DATA_FIELD_COUNT * BYTE_PER_FIELD) == 0;
+
+    // Send the packet to Flutter.
+    renderer.dispatchPointerDataPacket(packet, packet.position());
+
+    return true;
+  }
+
+  // TODO(mattcarroll): consider creating a PointerPacket class instead of using a procedure that mutates inputs.
+  private void addPointerForIndex(MotionEvent event, int pointerIndex, ByteBuffer packet) {
+    int pointerChange = getPointerChangeForAction(event.getActionMasked());
+    if (pointerChange == -1) {
+      return;
+    }
+
+    int pointerKind = getPointerDeviceTypeForToolType(event.getToolType(pointerIndex));
+
+    long timeStamp = event.getEventTime() * 1000; // Convert from milliseconds to microseconds.
+
+    packet.putLong(timeStamp); // time_stamp
+    packet.putLong(pointerChange); // change
+    packet.putLong(pointerKind); // kind
+    packet.putLong(event.getPointerId(pointerIndex)); // device
+    packet.putDouble(event.getX(pointerIndex)); // physical_x
+    packet.putDouble(event.getY(pointerIndex)); // physical_y
+
+    if (pointerKind == POINTER_DEVICE_KIND_MOUSE) {
+      packet.putLong(event.getButtonState() & 0x1F); // buttons
+    } else if (pointerKind == POINTER_DEVICE_KIND_STYLUS) {
+      packet.putLong((event.getButtonState() >> 4) & 0xF); // buttons
+    } else {
+      packet.putLong(0); // buttons
+    }
+
+    packet.putLong(0); // obscured
+
+    // TODO(eseidel): Could get the calibrated range if necessary:
+    // event.getDevice().getMotionRange(MotionEvent.AXIS_PRESSURE)
+    packet.putDouble(event.getPressure(pointerIndex)); // pressure
+    packet.putDouble(0.0); // pressure_min
+    packet.putDouble(1.0); // pressure_max
+
+    if (pointerKind == POINTER_DEVICE_KIND_STYLUS) {
+      packet.putDouble(event.getAxisValue(MotionEvent.AXIS_DISTANCE, pointerIndex)); // distance
+      packet.putDouble(0.0); // distance_max
+    } else {
+      packet.putDouble(0.0); // distance
+      packet.putDouble(0.0); // distance_max
+    }
+
+    packet.putDouble(event.getToolMajor(pointerIndex)); // radius_major
+    packet.putDouble(event.getToolMinor(pointerIndex)); // radius_minor
+
+    packet.putDouble(0.0); // radius_min
+    packet.putDouble(0.0); // radius_max
+
+    packet.putDouble(event.getAxisValue(MotionEvent.AXIS_ORIENTATION, pointerIndex)); // orientation
+
+    if (pointerKind == POINTER_DEVICE_KIND_STYLUS) {
+      packet.putDouble(event.getAxisValue(MotionEvent.AXIS_TILT, pointerIndex)); // tilt
+    } else {
+      packet.putDouble(0.0); // tilt
+    }
+  }
+
+  private int getPointerChangeForAction(int maskedAction) {
+    // Primary pointer:
+    if (maskedAction == MotionEvent.ACTION_DOWN) {
```

Comment:
The way you're mapping multiple types of motion events into POINTER_CHANGE_DOWN and POINTER_CHANGE_UP suggests we should list that in one of the caveats for this class.

If ACTION_UP and ACTION_POINTER_UP ever end up having different semantic meanings, those won't translate properly into Flutter.

Reply:
I think it'd be best to avoid changing this as part of this PR - introducing new changes like that should probably be done separately from this refactor.  It would be best to open a new issue for that.

Reply:
I don't suppose there's any way to have this history get tied to its origin in shell/platform/android/io/flutter/view/FlutterView.java?

Reply:
@dnfield I'm not aware of any way to link the history, but also git == magic...should I just add a comment at the top of the class that says this originally came from FlutterView?

Reply:
I guess just adding a comment at the top of the file or something would be helpful.  A move plus rename plus content change is tough to automatically track